docs/07_how-to/01_auth.mdx

```diff
@@ -0,0 +1,395 @@
+
+The code below looks up the value on the prior version and sets it on the proposed modification before saving, based on the Right Code **TradeViewFull**.
+
+```jsx
+  eventHandler<Trade>("TRADE_MODIFY", transactional = true) {
+    ...
+      if (!userHasRight(event.userName, "TradeViewFull")) {
+        val trade = entityDb.get(Trade.ById(event.details.tradeId))!!
+        details.customerName = trade.customerName
+      }
+      entityDb.modify(details)
+    ...
+  }
+```
+
+You can set up row-level authorisation through a further table where (continuing the example above) Countries are mapped to Profiles. This could be any reference field as per the authorisation requirements.
+
+Row-level permissioning is then performed via a look-up on an in-memory map (which is created and managed by the auth process):
+
+```jsx
+  query("ALL_TRADES", TRADE) {
+    permissioning {
+      auth(mapName = "COUNTRY_VISIBILITY") {
+        authKeyWithUserName {
+          key(data.countryName, userName)
+        }
+      }
+    }
+  }
+```
+
+You can specify row-level authorisation for each event in your Event Handler in a similar way:
+```jsx
+  eventHandler<Trade>("TRADE_MODIFY", transactional = true) {
+    permissioning {
+      auth(mapName = "COUNTRY_VISIBILITY") {
+        authKeyWithUserName {
+          key(data.countryName, userName)
+        }
+      }
+    }
+  }
+```
+
+You can combine table-, row- and column-level authorisation in a single data server.
+
+### Mass authorisation
+
+It is also possible to apply authorisation for all queries in a Data Server or all events in an Event Handler file. 
+
+When you do this, all queries or events within the .kts (file) receive these permissions. But you can override them within any specific query or event. So, effectively, you are providing default permission for all the queries or events in the file.
+
+This is useful, for example, where a project has collections of functions; you can create separate files of Data Servers or Event Handlers and simply apply the authorisation once in each file.
+
+Here is an example of a Data Server file, where the authorisation applies to every query in the file:
+
+
+```jsx
+dataServer {
+    permissioning {
+        permissionCodes = listOf("StaticUpdate")
+    }
+    query(...)
+    query(...)
+}
+```
+
+This works exactly the same way for an Event Handler file.
+
+All above patterns are the same for Request Replies (reqreps) within the Genesis Platform - this can be seen in the example application referenced below.
+
+### Front-end authorisation
+
+It is possible to apply authorisation at the front end to remove grids and buttons where there is no appropriate permission.  If this is not done, the system is protected (by the server) but users will receive error messages explaining their lack of permissions.
```

Comment:
could be seen as misleading to say it is protected. should we say "the system will be protected by any server permissions" ?